gitclone/checkout_simple_strategy.go

```diff
@@ -0,0 +1,129 @@
+package gitclone
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/bitrise-io/bitrise-init/step"
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+type checkoutStrategy interface {
+	Validate() error
```

Comment:
Why do we have a separate validation?

The checkout strategy is selected based on the checkout params. if commit != "", then commitCheckout.